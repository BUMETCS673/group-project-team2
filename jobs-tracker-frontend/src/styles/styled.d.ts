import 'styled-components';


declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
        nav :{
            backgroundColor: string;
    }
      primary: {
          main: string;
    }
      secondary: {
          main: string;
          dark: string;
      }
      text: {
          main: string;
          secondary: string;
      }
      background: {
        primary: string;
        secondary: string;
      }
    
    }
    input: {
        borderRadius: string;
        backgroundColor: {
            primary: string;
            secondary: string;

        },
        opacity: number;
    }
   
  }
}