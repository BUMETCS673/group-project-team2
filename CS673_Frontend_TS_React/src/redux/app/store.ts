import { configureStore } from '@reduxjs/toolkit'
import jobReducer from '../slices/model/job'
import userReducer from '../slices/model/user'
import activityReducer from '../slices/model/activity'
import uiReducer from '../slices/model/ui'
import { apiSlice } from '../slices/api/jobs'
import { activitySlice } from '../slices/api/activities'

export const store = configureStore({
  reducer: {
    job: jobReducer,
    user: userReducer,
    activity: activityReducer,
    ui: uiReducer,
    [apiSlice.reducerPath]: apiSlice.reducer,
    [activitySlice.reducerPath]: activitySlice.reducer,
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware().concat(
      apiSlice.middleware,
      activitySlice.middleware
    )
  },
})
export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
