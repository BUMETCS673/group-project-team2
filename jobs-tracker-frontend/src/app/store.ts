import {configureStore} from '@reduxjs/toolkit'
import jobReducer from "../features/job/job-slice"
import userReducer from "../features/user/user-slice"


export const store = configureStore({
    reducer: {
        job: jobReducer,
        user: userReducer,
    }
})
export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>