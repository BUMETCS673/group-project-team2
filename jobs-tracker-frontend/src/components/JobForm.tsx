import {Formik, Field, Form, FormikHelpers} from 'formik'
interface Values {
    companyName: string;
    jobTitle: string;
}
const sleep = ms => new Promise(r => setTimeout(r, ms))
const JobForm = ({onSubmit}) => {
    const handleSubmit = async values= => {
        await sleep(500)
        onSubmit(values)
      }
    <div>
      <Formik
        initialValues={{
          companyName: '',
          jobTitle: '',
        }}
        onSubmit={handleSubmit}
      >
        <Form>
          <label htmlFor="companyName">Company</label>
          <Field id="companyName" name="companyName" />

          <label htmlFor="jobTitle">Last Name</label>
          <Field id="jobTitle" name="jobTitle" />

          
          <button type="submit">Save</button>
        </Form>
      </Formik>
    </div>
}
export default JobForm