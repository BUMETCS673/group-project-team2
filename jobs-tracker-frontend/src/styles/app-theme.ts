import { DefaultTheme } from "styled-components";

const appTheme: DefaultTheme = {
  colors: {
      nav: {
          backgroundColor:"rgba(0, 0, 0, 0.83)",
      },
    primary: {
      main: "#80D0C2",
    },
    secondary: {
      main: "#EF9C4B",
      dark: "#FD8030",
    },
    text: {
      main: "#848080",
      secondary: "#FFFFFF",
    },
    background: {
      primary: "rgba(128, 208, 194, 0.1)",
      secondary: "#F9F9F9",
    }
  },
  input: {
    borderRadius: "10px",
    backgroundColor: {
      primary: "#F8F8F8",
      secondary: 'rgba(239, 223, 75, .23)',
    },
    opacity: 0.4,
  },

};

export { appTheme };
