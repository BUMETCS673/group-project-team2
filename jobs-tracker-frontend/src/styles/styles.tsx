import styled from 'styled-components'
import { NavLink } from "react-router-dom"

export const button = styled.button`
  border: 1px dotted yellow;
  background: white;
  color: black;
  font-weight: 500;
`

export const Container = styled.div`
  border: 2px solid blue;
  width: 100px;
`
interface MenuItemsProps {
  readonly menu?: boolean;
}
export const FlexContainer = styled.div<MenuItemsProps>`
  display: flex;
  flex-direction: column;
  ${(props) => (props.menu && `
  @media (min-width: 768px) {
    flex-direction: row;
  }
  `)}

`
export const StyledNav = styled.nav`
display: flex;
justify-content: space-between;
height: 60px;
background-color: ${(props) => props.theme.colors.nav.backgroundColor};
box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
opacity: 1;

`
export const StyledPrimaryBtn = styled.button<MenuItemsProps>`
  border-radius: 20px;
  font-size: 20px;
  padding: 5px 30px;
  font-weight: bold;
  background-color: ${props => props.theme.colors.primary.main};
  color: ${props => props.theme.colors.text.secondary};
  border: none;
  ${props => (props.menu ? `
  margin: 0px;
  @media (min-width: 768px) {
    margin: 0 20px;
  }
  `: `
  margin: 10px;
  `)};
`

export const SimpleButton = styled.button`
background-color: transparent;
color: ${props => props.theme.colors.primary.main};
font-size: 20px;
text-align: left;
border: none;
&:hover {
      font-size: 25px;
      background-color: transparent;
      opacity: 0.5;
  }
  @media (min-width: 768px) {
    margin-left: 10px;
  }

`

export const Logo = styled(NavLink)`
text-align: center;
margin-left: 20px;
width: 200px;
img {
  width: 90%;
}
`;
export const Collapse = styled.div`
    display: none;
    @media (min-width: 768px) {
        display: flex;
    }
`
export const StyledMenuButton = styled.button`
    background-color: transparent;
    display: flex;
    align-self:center;
    border: 2px solid ${props => props.theme.colors.text.main};
    color: ${props => props.theme.colors.text.main};
    padding: 0px;
    border-radius: 10px;
    margin-right: 30px;
    &:hover {
      background-color: transparent;
      color: ${props => props.theme.colors.text.secondary};
      border-color: ${props => props.theme.colors.text.secondary};
      opacity: 0.9;
    }
    @media (min-width: 768px) {
        display: none;
    }
`
export const MenuLinkGroup = styled.div`
display: flex;
flex-direction: column;
align-items: start;
padding: 20px 0 40px 0;
margin: 0 15px;
border-bottom: 1px solid rgba(255, 255, 255, 0.25);
@media (min-width: 768px) {
    flex-direction: row;
    align-items: center;
    padding: 0;
    border-bottom: none;
}
`
export const StyledNavLink = styled(NavLink)`
  font-size: 20px;
  color: ${props => props.theme.colors.text.secondary};
  opacity: 0.25;
  text-decoration: none;
  padding: 10px 0;
  &.active {
      opacity: 1;
  }
  @media (min-width: 768px) {
    margin: 0 20px;
    padding: 0;
  }
  
`;
export const UserContainer = styled.div`
  display: flex;
  color: ${props => props.theme.colors.text.secondary};
  align-items: center;
  margin: 0px;
  padding: 0px;
  img {
    width: 30px;
    height: 30px;
    border-radius: 50%;
    margin-right: 10px;
  }
  p {
    opacity: 0.25;
    font-size: 20px;
    width: 110px;
  }

  
`
interface SidebarProps {
  readonly toggleSidebar?: boolean;
}

export const StyledSidebar =styled.div<SidebarProps>`
display: ${(props) => (props.toggleSidebar ? "flex" : "none")};
flex-direction: column;
width: 250px;
height: 100vh;
position: fixed;
top: 0;
right: 0;
padding: 2rem 1rem 1rem 1rem;
background-color: #1a1a1a;
z-index: 9999;
@media (min-width: 768px) {
    display: none;
}
`;

export const StyledCloseButton = styled.button`
    background-color: transparent;
    border: none;
    color: ${props => props.theme.colors.text.main};
    align-self: flex-end;
    position: absolute; top: 10px; right: 10px;
    &:hover {
      background-color: transparent;
      color: ${props => props.theme.colors.text.secondary};

  }
`
export const StyledCardsContainer = styled.div`
  margin: auto;
  background-color: ${props => props.theme.colors.background.secondary};
  height: 100vh;
  padding: 20px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  @media (min-width: 768px) {
    width: 70%;
  }
`
export const StyledCardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  border: 1px solid red;
  border-radius: 5px;
  align-items: center;
  justify-content: center;
  position: relative;

`
export const StyledDeleteBtn = styled.button`
background-color: transparent;
border: none;
border-radius: 10px;
color: ${props => props.theme.colors.text.main};
align-self: flex-end;
position: absolute; top: 10px; right: 10px;
&:hover {
  background-color: ${props => props.theme.colors.primary.main} ;
  color: ${props => props.theme.colors.text.secondary};
}
`
export const StyledCardHeader = styled.div`
  display: flex;


`
export const StyledCardBody = styled.div`
  h1 {
    font-size: 26px;

  }
  h2 {
    font-size: 20px;
    color: ${props => props.theme.colors.text.main};
    
  }

`
export const StyledCardFooter = styled.div`
  display: flex;
  justify-content: space-between;
  p {
    margin: 0 10px;
  }

`

