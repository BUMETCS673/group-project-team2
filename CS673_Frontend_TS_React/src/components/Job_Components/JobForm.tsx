import { Formik, FormikHelpers, FormikErrors, Field } from 'formik'
import { Row, Col, Form, FormSubtitle, HelperText } from '../../styles/styles'
import { useAppDispatch, useAppSelector } from '../../app/hooks'
import { setJob } from '../../features/job/job-slice'
import { useCreateJobMutation, useUpdateJobMutation } from '../../features/jobs/jobs-api-slice'
import { Button } from '@mui/material'
import CheckIcon from '@mui/icons-material/Check'
import { Job } from '../../types/types'
// import { useHistory } from 'react-router-dom'

interface Values {
  companyname: string
  jobtitle: string
  description: string
  status: string
}

interface JobFormType {
  job_id?: string | undefined
  closePopup: () => void
}

const JobForm: React.FC<JobFormType> = ({
  job_id,
  closePopup,
}: JobFormType) => {
  // const { push } = useHistory()
  const dispatch = useAppDispatch()
  const jobsList = useAppSelector(state => state.user.jobs)
  const jobToEdit:Job | null = jobsList.filter( job => job.ID == job_id)[0]
  const [createJob, newJob] = useCreateJobMutation()
  const [updateJob, editJob] = useUpdateJobMutation()
  console.log(newJob, job_id)
  console.log(editJob, job_id)
  //const job = useAppSelector(state => state.job)

  return (
    <div>
      <Formik
        initialValues={{
          companyname: jobToEdit ? jobToEdit.companyname : '',
          jobtitle: jobToEdit ? jobToEdit.jobtitle : '',
          description: jobToEdit ? jobToEdit.description : '',
          status: jobToEdit ? jobToEdit.status : '',
        }}
        validate={(values: Values) => {
          let errors: FormikErrors<Values> = {}
          if (!values.companyname) {
            errors.companyname = 'Required'
          }
          if (!values.jobtitle) {
            errors.jobtitle = 'Required'
          }
          if (!values.status) {
            errors.status = 'Required'
          }
          return errors
        }}
        onSubmit={async (
          values: Values,
          { setSubmitting }: FormikHelpers<Values>
        ) => {
          console.log(values)
          if (jobToEdit) {
            await updateJob(values)
          } else {
            dispatch(setJob(values))
            await createJob(values)
          }
          setSubmitting(false)
          closePopup()
          // push('/home')
        }}
      >
        {({
          handleSubmit,
          handleChange,
          handleBlur,
          touched,
          errors,
          values,
          isValid,
          dirty,
        }) => (
          <Form onSubmit={handleSubmit}>
            <FormSubtitle>Job info</FormSubtitle>

            <Row>
              <Col>
                <Field
                  onChange={handleChange}
                  onBlur={handleBlur}
                  value={values.companyname}
                  //border = {!(touched.companyname  && errors.companyname && "1px solid red" )}
                  id="companyname"
                  data-testid="companyname"
                  name="companyname"
                  placeholder="Company"
                />
                {touched.companyname && errors.companyname && (
                  <HelperText data-testid="companynameError">{errors.companyname}</HelperText>
                )}
              </Col>
              <Col>
                <Field
                  onChange={handleChange}
                  onBlur={handleBlur}
                  value={values.jobtitle}
                  //border = {!(touched.jobtitle  && errors.jobtitle && "1px solid red")}
                  id="jobtitle"
                  name="jobtitle"
                  data-testid="jobtitle" 
                  placeholder="Job Title"
                />
                {touched.jobtitle && errors.jobtitle && (
                  <HelperText data-testid="jobtitleError">{errors.jobtitle}</HelperText>
                )}
              </Col>
            </Row>
            <Row>
              <Col>
                <Field
                  as="textarea"
                  onChange={handleChange}
                  onBlur={handleBlur}
                  value={values.description}
                  id="description"
                  name="description"
                  placeholder="Job description..."
                />
              </Col>

              <Col>
                <Field
                  name="status"
                  onChange={handleChange}
                  onBlur={handleBlur}
                  placeholder="Status"
                  data-testid = "status"
                  value={values.status}
                ></Field>
                {/* <Field 
                      component = "select"
                      onChange = {handleChange}
                      
                      //border = {!(touched.status  && errors.status && "1px solid red")}
                       
                      name="status" 
                      
                      
                    >
                      <option value = "in progress">In progress</option>
                      <option  value = "completed process">Completed process</option>
                      <option value = "received offer">Received Offer</option>
                    </Field> */}
                {touched.status && errors.status && (
                  <HelperText data-testid="statusError">{errors.status}</HelperText>
                )}
              </Col>
            </Row>

            <Button type="submit" disabled={!isValid || !dirty}>
              <CheckIcon fontSize="large" /> Save job
            </Button>
          </Form>
        )}
      </Formik>
    </div>
  )
}
export default JobForm
