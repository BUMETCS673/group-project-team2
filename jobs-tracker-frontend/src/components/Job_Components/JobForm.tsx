import { Formik, FormikHelpers, FormikErrors, Field } from 'formik'
import { Row, Col, Form, FormSubtitle, HelperText } from '../../styles/styles'
import { useAppDispatch } from '../../app/hooks'
import { setJob } from '../../features/job/job-slice'
import { useCreateJobMutation } from '../../features/jobs/jobs-api-slice'
// import { useHistory } from 'react-router-dom'

interface Values {
  companyname: string
  jobtitle: string
  description: string
  status: string
}

interface JobFormType {
  job_id?: string | undefined
  closePopup: () => void
}

const JobForm: React.FC<JobFormType> = ({
  job_id,
  closePopup,
}: JobFormType) => {
  // const { push } = useHistory()
  const dispatch = useAppDispatch()
  const [createJob, data] = useCreateJobMutation()
  console.log(data, job_id)
  //const job = useAppSelector(state => state.job)

  return (
    <div>
      <Formik
        initialValues={{
          companyname: '',
          jobtitle: '',
          description: '',
          status: '',
        }}
        validate={(values: Values) => {
          let errors: FormikErrors<Values> = {}
          if (!values.companyname) {
            errors.companyname = 'Required'
          }
          if (!values.jobtitle) {
            errors.jobtitle = 'Required'
          }
          if (!values.status) {
            errors.status = 'Required'
          }
          return errors
        }}
        onSubmit={async (
          values: Values,
          { setSubmitting }: FormikHelpers<Values>
        ) => {
          console.log(values)
          dispatch(setJob(values))
          await createJob(values)
          setSubmitting(false)
          closePopup()
          // push('/home')
        }}
      >
        {({
          handleSubmit,
          handleChange,
          handleBlur,
          touched,
          errors,
          values,
          isValid,
          dirty,
        }) => (
          <Form onSubmit={handleSubmit}>
            <FormSubtitle>Job info</FormSubtitle>

            <Row>
              <Col>
                <Field
                  onChange={handleChange}
                  onBlur={handleBlur}
                  value={values.companyname}
                  //border = {!(touched.companyname  && errors.companyname && "1px solid red" )}
                  id="companyname"
                  name="companyname"
                  placeholder="Company"
                />
                {touched.companyname && errors.companyname && (
                  <HelperText>{errors.companyname}</HelperText>
                )}
              </Col>
              <Col>
                <Field
                  onChange={handleChange}
                  onBlur={handleBlur}
                  value={values.jobtitle}
                  //border = {!(touched.jobtitle  && errors.jobtitle && "1px solid red")}
                  id="jobtitle"
                  name="jobtitle"
                  placeholder="Job Title"
                />
                {touched.jobtitle && errors.jobtitle && (
                  <HelperText>{errors.jobtitle}</HelperText>
                )}
              </Col>
            </Row>
            <Row>
              <Col>
                <Field
                  as="textarea"
                  onChange={handleChange}
                  onBlur={handleBlur}
                  value={values.description}
                  id="description"
                  name="description"
                  placeholder="Job description..."
                />
              </Col>

              <Col>
                <Field
                  name="status"
                  onChange={handleChange}
                  placeholder="Status"
                ></Field>
                {/* <Field 
                      component = "select"
                      onChange = {handleChange}
                      
                      //border = {!(touched.status  && errors.status && "1px solid red")}
                       
                      name="status" 
                      
                      
                    >
                      <option value = "in progress">In progress</option>
                      <option  value = "completed process">Completed process</option>
                      <option value = "received offer">Received Offer</option>
                    </Field> */}
                {touched.status && errors.status && (
                  <HelperText>{errors.status}</HelperText>
                )}
              </Col>
            </Row>
            <FormSubtitle>Activities</FormSubtitle>
            <Row>
              <Col></Col>
            </Row>

            <button type="submit" disabled={!isValid || !dirty}>
              Save
            </button>
          </Form>
        )}
      </Formik>
    </div>
  )
}
export default JobForm
