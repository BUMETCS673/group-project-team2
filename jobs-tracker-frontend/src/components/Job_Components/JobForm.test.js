import React from 'react';
import { render, screen, waitFor, fireEvent} from '@testing-library/react'
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom'
import JobForm from './JobForm';
import {Provider} from 'react-redux' 
import {store} from '../../app/store'
import { appTheme } from '../../styles/app-theme'
import { ThemeProvider } from 'styled-components'



describe("JobForm component", () => {
    it('renders without crashing', () => {
        const component = renderer.create(
            <Provider store = {store}>
                <ThemeProvider theme = {appTheme}>
                    <JobForm/>
                </ThemeProvider>
                
            </Provider>
            
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    });
    
    it('renders and submits a formik form', async() => {
        const handleSubmit = jest.fn()
        render(
        <Provider store = {store}>
            <ThemeProvider theme = {appTheme}>
                <JobForm onSubmit={handleSubmit} />
            </ThemeProvider>
            
        </Provider>)
        

        await fireEvent.input(screen.getByTestId('companyname', 'Amazon'))
        await fireEvent.input(screen.getByTestId('jobtitle', 'Software Engineer'))
        await fireEvent.input(screen.getByTestId( 'status', 'in progress'))

        await fireEvent.click(screen.getByRole('button', {name: /save/i}))

        await waitFor(() =>
            expect(handleSubmit).toHaveBeenCalledWith({
            companyname: 'Amazon',
            jobtitle: 'Software Engineer',
            status: 'in progress',
            }),
        )
      })
    
})