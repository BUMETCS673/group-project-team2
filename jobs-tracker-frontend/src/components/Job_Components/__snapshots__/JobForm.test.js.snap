// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobForm component renders without crashing 1`] = `
<div>
  <form
    className="sc-eCYdqJ eMRzOJ"
    onSubmit={[Function]}
  >
    <h3
      className="sc-hKMtZM HGvDf"
    >
      Job info
    </h3>
    <div
      className="sc-dkzDqf djoXzZ"
    >
      <div
        className="sc-jSMfEi ezyuOj"
      >
        <input
          id="companyname"
          name="companyname"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Company"
          value=""
        />
      </div>
      <div
        className="sc-jSMfEi ezyuOj"
      >
        <label
          className="sc-himrzO bBCmOf"
        >
          <input
            id="jobtitle"
            name="jobtitle"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Job Title"
            value=""
          />
        </label>
      </div>
    </div>
    <div
      className="sc-dkzDqf djoXzZ"
    >
      <div
        className="sc-jSMfEi ezyuOj"
      >
        <label
          className="sc-himrzO bBCmOf"
        >
          <textarea
            id="description"
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Job description..."
            value=""
          />
        </label>
      </div>
      <div
        className="sc-jSMfEi ezyuOj"
      >
        <label
          className="sc-himrzO bBCmOf"
        >
          <input
            name="status"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Status"
            value=""
          />
        </label>
      </div>
    </div>
    <h3
      className="sc-hKMtZM HGvDf"
    >
      Activities
    </h3>
    <div
      className="sc-dkzDqf djoXzZ"
    >
      <div
        className="sc-jSMfEi ezyuOj"
      />
    </div>
    <button
      disabled={true}
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;
